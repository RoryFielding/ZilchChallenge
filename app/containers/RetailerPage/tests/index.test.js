/**
 * Test the HomePage
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { HomePage } from '../index';
import configureStore from '../../../configureStore';

describe('<RetailerPage />', () => {
  let store;
  const location = {
    pathname: '/amazon',
  };
  const onInit = jest.fn();

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <HomePage
            loading={false}
            error={false}
            retailers={[]}
            location={location}
            onInit={onInit}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should fetch the retailers on mount', () => {
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <HomePage
            loading={false}
            error={false}
            retailers={[]}
            location={location}
            onInit={onInit}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(onInit).toHaveBeenCalled();
  });
});
