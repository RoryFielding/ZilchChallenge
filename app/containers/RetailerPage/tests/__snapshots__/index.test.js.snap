// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RetailerPage /> should render and match the snapshot 1`] = `
<article>
  <div
    class="Container-sc-7ubm28-0 bbWRub"
  />
</article>
`;
